library/ssl_tls13_generic.c

```diff
@@ -966,6 +982,143 @@ int mbedtls_ssl_tls13_write_certificate( mbedtls_ssl_context *ssl )
 /*
  * STATE HANDLING: Output Certificate Verify
  */
+static int ssl_tls13_get_sig_alg_from_pk( mbedtls_ssl_context *ssl,
```

Comment:
Actually there is another way that is probably better I'd say: introduce an
```C
int ssl_tls13_get_pk_type_and_md_alg_from_sig_alg(
    uint16_t sig_alg, mbedtls_pk_type_t *pk_type, mbedtls_md_type_t *md_alg)
```
utility function that contains the code in `ssl_tls13_parse_certificate_verify` between lines 339 and 378. Call this in `ssl_tls13_parse_certificate_verify` and `ssl_tls13_write_certificate_verify_body`. We would have a better separation of tasks with overall less code.